KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
+    """
+    This section checks through the Addendum and identifies words that are valid keywords of sections
+    It then takes that list and removes duplicates, as well as printing the output to another file.
+
+    :param kls: The addendum that provides the words.
+    :return: Inputs results into a global list.
+    """
+
+    kls_list_chap = []
+    for i in kls:
+        if kls.index(i) > 313:
+            if "paragraph{" in i or "subsubsection*{" in i:
+                kls_list.append(i[i.find("{") + 1: len(i) - 2])
+
+    i = 0
+    while i < len(kls_list):
+        if "reference" in kls_list[i].lower() or "limit relation" in kls_list[i].lower():
+
+            del kls_list[i]
+        else:
+            i += 1
+    kls_list.append("Limit Relation")
+    for i in kls_list:
+        add = True
+        for j in kls_list_chap:
+            if i == j:
+                add = False
+        if add:
+            kls_list_chap.append(i)
+    return kls_list_chap
 
 
-def prepareForPDF(chap):
-    footmiscIndex = 0
+def fix_chapter_sort(kls, chap, word, sortloc):
+    """
+    This function sorts through the input files and identifies and places sections from the addendum after their
+    respective subsections in the chapter.
+
+    :param kls: The addendum where the sections to be inserted are found.
+    :param chap: The destination of inserted sections.
+    :param word: The keyword that is being processed.
+    :param sortloc: The location in the list of words, of the word being sorted.
+    :return: This function outputs the processed chapter into a larger function for further processing.
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    sep1 = 0
+    name_chap = word.lower()
+    if name_chap == "orthogonality":
+        sep1 = 1
+    elif name_chap == "special value":
+        sep1 = 2
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    for i in kls:
+        index += 1
+        line = str(i)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12)+1: line.find("}", 12)]  # get just the name (like mathpeople)
+
+        if sep1 == 0:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line):
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 1:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line) and "orthogonality relation" not in line.lower():
+                for i in klsaddparas:
+                    if index < i:
```

Comment:
These variables should be named differently to make it clear what the difference is and why they may be compared.